src/test/java/ch/qos/logback/more/appenders/LogbackAppenderTest.java

```diff
@@ -74,4 +76,18 @@ public void logThrowable() throws InterruptedException {
 
         Thread.sleep(1000); // Wait a moment because these log is being appended asynchronous...
     }
+
+    @Test
+    public void logMarkerMap() throws InterruptedException {
+
+        Map<String, String> map = new HashMap<String, String>();
+        map.put("key1", "value1");
+        map.put("key2", "value2");
+
+        MapMarker markerMap = new MapMarker("MAP_MARKER", map);
+
+        LOG.debug(markerMap, "Test the marker map.");
```

Comment:
@mahanhz Sorry for the late reply. 
Can I modify it like this?  `marker.MAP_MARKER":{"key1":"value1","key2":"value2"}`. Originally the map was flatten as top level but I think it's better to specify it as a marker.
```json
{"msg":"22:30:24.676 [main] DEBUG c.q.l.m.a.LogbackAppenderTest#? Test the marker map.","caller":"?#?:?\n","level":"DEBUG","logger":"ch.qos.logback.more.appenders.LogbackAppenderTest","foo":"bar","marker.MAP_MARKER":{"key1":"value1","key2":"value2"},"thread":"main","message":"Test the marker map.","foo2":"bar2"}
```

Reply:
@sndyuk @mahanhz Thanks for your efforts on this project. It has been very useful! 
I have been watching this PR/issue, and it would actually help me out more if the map entries were inserted at the top level (flat). Is this perhaps something we could configure at the global level? (e.g. `MAP_MARKER_FLATTEN = true/false`)?

Reply:
@zonybob Thank you for the good opinion. I added the `flattenMapMarker` option. 

Also I modified it as the mapMarker can be nested. https://github.com/sndyuk/logback-more-appenders/pull/32/files#diff-0f8e83f64595674bdec3131517e62d16R95